src/infra/core/src/pull_service_impl.rs

```diff
@@ -447,6 +447,37 @@ impl PullServiceImpl {
             })
             .collect())
     }
+
+    /// Update summary for every updated dataset
+    async fn update_summaries(&self, results: &Vec<PullResponse>) {
+        for res in results {
+            if let Ok(r) = &res.result {
+                if let Some(local) = &res.local_ref {
+                    match r {
+                        PullResult::Updated { .. } => {
+                            match self.dataset_repo.clone().find_dataset_by_ref(local).await {
+                                Ok(ds) => {
+                                    if let Err(e) = ds
+                                        .clone()
+                                        .get_summary(GetSummaryOpts {
+                                            update_if_stale: true,
+                                        })
+                                        .await
+                                    {
+                                        tracing::error!("Failed to update dataset summary: {}", e);
+                                    }
+                                }
```

Comment:
You are not returning any error in this scenario, but this seems like an internal failure.